struqture/src/mixed_systems/mixed_hermitian_product.rs

```diff
@@ -281,81 +332,92 @@ impl MixedIndex for HermitianMixedProduct {
         let spins: TinyVec<[PauliProduct; 2]> = spins.into_iter().collect();
         let bosons: TinyVec<[BosonProduct; 2]> = bosons.into_iter().collect();
         let fermions: TinyVec<[FermionProduct; 2]> = fermions.into_iter().collect();
-        let (new_index, new_val) = if let Some(b) = bosons.iter().next() {
-            if b.creators().min() > b.annihilators().min() {
-                let mut new_spins: TinyVec<[PauliProduct; 2]> = TinyVec::<[PauliProduct; 2]>::new();
-                let mut prefactor = 1.0;
-                for s in spins {
-                    let (new_s, pf) = s.hermitian_conjugate();
-                    new_spins.push(new_s);
-                    prefactor *= pf;
-                }
-                let mut new_bosons: TinyVec<[BosonProduct; 2]> =
-                    TinyVec::<[BosonProduct; 2]>::new();
-                for b in bosons {
-                    let (new_b, pf) = b.hermitian_conjugate();
-                    new_bosons.push(new_b);
-                    prefactor *= pf;
-                }
-                let mut new_fermions: TinyVec<[FermionProduct; 2]> =
-                    TinyVec::<[FermionProduct; 2]>::new();
-                for f in fermions {
-                    let (new_f, pf) = f.hermitian_conjugate();
-                    new_fermions.push(new_f);
-                    prefactor *= pf;
+
+        let mut hermitian_conjugate = false;
+        let mut hermitian_decision_made = false;
+        for b in bosons.iter() {
+            let mut number_equal_indices = 0;
+            for (creator, annihilator) in b.creators().zip(b.annihilators()) {
+                match annihilator.cmp(creator) {
+                    std::cmp::Ordering::Less => {
+                        hermitian_conjugate = true;
```

Comment:
Should we also have `hermitian_decision_made = true` and `break` in this arm?

Reply:
Yes